### PR TITLE
Enhance some stuff

### PR DESCRIPTION
**Pull request by FearlessSpiff**
* added deletion of uploaded files
* added setting to create a unique target dir or not
* added setting to configure a target dir
* made some refactorings
* fixed crash. same as in STGUploader commit 5a6681f

### CODE REVIEW

File: app/src/main/java/org/bostwickenator/ftpuploader/SettingsStore.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
     private JSONObject settings;
     private final File settingsFile = FileGetter.getFile("SET.JSN");
 
-    public static SettingsStore getSettingsStore() {
+    public static SettingsStore settingsStore() {
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/SettingsStore.java:25]: Not a fan of this naming refactor, it creates the potential to confuse you, suddenly you might be doing `new settingsStore()` not realizing this is a  monolithic pattern.

A more ideal name would be the previous or even `getSettingsStoreInstance()`.

File: app/src/main/java/org/bostwickenator/ftpuploader/WifiSetupActivity.java [hunk 2]
```diff
@@ -123,7 +123,6 @@ private void updateWifiSwitch() {
                         summary = res.getString(R.string.connectionStateConnecting);
                         break;
                     default:
-                        res.getString(0, "test");
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/WifiSetupActivity.java:126]: Patch from Bostwickenator/STGUploader@5a6681f should be in it's own PR

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java [hunk 3]
```diff
@@ -136,9 +142,10 @@ public void run() {
                     }
                 });
 
-                String albumName = "FTP_Uploader_" + DateUtils.getDate();
-
-                ftp.createAndUseAlbumForSubsequentOperations(albumName);
+                if (settingsStore().getBoolean(SETTING_CREATE_ALBUM, true)) {
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:146]: Create a variable for the settings store, don't repeatedly call the instance for no good reason

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java [hunk 4]
```diff
@@ -149,6 +156,11 @@ public void run() {
                     boolean result = ftp.storeFile(file);
                     if(result) {
                         uploadRecordDatabase.addFile(file);
+                        if (settingsStore().getBoolean(SETTING_DELETE_AFTER_UPLOAD, false)) {
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:160]: see previous comment re settingsstore refactoring

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java [hunk 5]
```diff
@@ -167,7 +179,7 @@ public void run() {
     private List<File> getFilesToUpload(){
         List<File> files = FilesystemScanner.getImagesOnExternalStorage();
 
-        if(SettingsStore.getSettingsStore().getBoolean(SettingsActivity.SETTING_UPLOAD_VIDEOS, false)) {
+        if(settingsStore().getBoolean(SETTING_UPLOAD_VIDEOS, false)) {
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:185]: see previous comment re settingsstore refactoring

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java [hunk 6]
```diff
@@ -14,6 +14,12 @@
 import com.github.ma1co.pmcademo.app.BaseActivity;
 import com.github.ma1co.pmcademo.app.Logger;
 
+import static org.bostwickenator.ftpuploader.DateUtils.getDate;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_CREATE_ALBUM;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_DELETE_AFTER_UPLOAD;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_UPLOAD_VIDEOS;
+import static org.bostwickenator.ftpuploader.SettingsStore.settingsStore;
+
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:23]: Don't statically import constants, it's frowned upon normally.

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java [hunk 7]
```diff
@@ -9,43 +9,58 @@
...
         FileInputStream fis = new FileInputStream(file);
-        boolean result = ftpClient.storeFile(file.getName(), fis);
+        boolean result = this.ftpClient.storeFile(file.getName(), fis);
         fis.close();
         return result;
     }
 
     public void disconnect() throws IOException {
-        ftpClient.disconnect();
+        this.ftpClient.disconnect();
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:64]: Appending `this` isn't strictly necessary here, it should really only be used in cases where you want to assign a property from a constructor parameter.

Shame automatic assignment of constructor parameters to the relevant property isn't a thing anyway in java..  Maybe Java should learn from Typescript? ;)  [/unrelated rant]

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java [hunk 8]
```diff
@@ -9,43 +9,58 @@
...
 
+import static java.lang.Integer.parseInt;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_PASSIVE;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_PASSWORD;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_PORT;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_SERVER;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_TARGET_DIR;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_USERNAME;
+import static org.bostwickenator.ftpuploader.SettingsStore.settingsStore;
+
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:20]: Importing constants statically is frowned upon, should be used sparingly.

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java [hunk 9]
```diff
@@ -9,43 +9,58 @@
...
 class FtpClient {
 
     private final FTPClient ftpClient = new FTPClient();
     private final static String EMPTY_STRING = "";
-    SettingsStore settingsStore = SettingsStore.getSettingsStore();
+    private SettingsStore settingsStore = settingsStore();
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:25]: See other comment re name refactoring

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java [hunk 10]
```diff
@@ -149,6 +156,11 @@ public void run() {
                     boolean result = ftp.storeFile(file);
                     if(result) {
                         uploadRecordDatabase.addFile(file);
+                        if (settingsStore().getBoolean(SETTING_DELETE_AFTER_UPLOAD, false)) {
+                            if (!file.delete()) {
+                                Logger.error("failed deleting file");
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java]: Is it possible to provide the logger with which file failed to delete? 
Would help with troubleshooting for the user.

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java [hunk 11]
```diff
@@ -9,43 +9,58 @@
...
 
     public void connect() {
 
         try {
-            ftpClient.connect(getStringWithEmptyAsDefault(SettingsActivity.SETTING_SERVER), Integer.parseInt(getStringWithEmptyAsDefault(SettingsActivity.SETTING_PORT)));
-            ftpClient.login(getStringWithEmptyAsDefault(SettingsActivity.SETTING_USERNAME), getStringWithEmptyAsDefault(SettingsActivity.SETTING_PASSWORD));
-            if (settingsStore.getBoolean(SettingsActivity.SETTING_PASSIVE, false)) {
-                ftpClient.enterLocalPassiveMode();
+            this.ftpClient.connect(getStringWithEmptyAsDefault(SETTING_SERVER), parseInt(this.settingsStore.getString(SETTING_PORT, "21")));
```
Comment (Deph0) [inline app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:30]: Couldn't you just get this as a integer from the start or does the store not provide a .getInt() ?

**Review by Deph0 — CHANGES_REQUESTED** [review]
Great job extending the features to improve this project, however i added some review comments, minor things should be changed, but the more noteworthy is the patch from STGUploader- it should be moved to it's own PR.

I think you went a little bit too much with the refactoring, but if you had kept it simple this would have been a great PR.

**Comment by FearlessSpiff** [comment]
First, thanks for your input. I also think doing an own pr for the bugfix would have been better. but I was just too lazy after testing my code and finding this bug and even finding a fix for it. (Wasn't that easy to check alt hose repos.) I justput it in a own commit thereof. The rest I willanwer directly in the PR.

File: app/src/main/java/org/bostwickenator/ftpuploader/SettingsStore.java (hunk 1 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/SettingsStore.java:25]: Using get for getters is somehow old baken. I find it much more readable. Think this is preferences. Instantiating it with new () is not possible as the constructor is private...

File: app/src/main/java/org/bostwickenator/ftpuploader/WifiSetupActivity.java (hunk 2 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/WifiSetupActivity.java:126]: Would be better, true.

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java (hunk 3 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:146]: Compiler optimizes this and is much more readable. Just my opinion though. Against yours. ;-)

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java (hunk 4 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:160]: dito

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java (hunk 5 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java:185]: dito

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java (hunk 7 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:64]: Also matter of preference. Better than to use i.e. mFtpClient as somewhere else in the code base. Ilike it because I can see on one shot that this is an instance var and not something else defined in the scope of the method.

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java (hunk 8 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:20]: Also 1990 style. ;-) It is much more readable like this if the static constant expresses enough to be understand, as it is the case here.

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java (hunk 9 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:25]: True. I thought about removing this line of code and using settingsStore() everywhere else instead. But thought this to be too offensive. :-) But would like it better though...

File: app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java (hunk 10 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/MainActivity.java]: Good idea. Done in 29f3d1edc85efd0c6abe8a104f76899b3a6207e2

File: app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java (hunk 11 above)
Comment (FearlessSpiff) [reply app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java:30]: Didn't change that. Is from original code. Could surely be refactored also...

**Review by Deph0 — APPROVED** [review]

**Comment by Bostwickenator** [comment]
Hey guys thanks for the work. I was out for thanksgiving but I'll look this over shortly and get a new version out.

**Review by Bostwickenator — APPROVED** [review]
